Sources/NIO/SocketChannel.swift

```diff
@@ -789,7 +791,10 @@ final class SocketChannel: BaseSocketChannel<Socket> {
     }
 
     fileprivate init(socket: Socket, parent: Channel? = nil, eventLoop: SelectableEventLoop) throws {
-        try socket.setNonBlocking()
+        #if !os(Linux)
+            // Linux uses accept4 with SOCK_NONBLOCK flag to get a nonblocking socket back from accept
+            try socket.setNonBlocking()
+        #endif
```

Comment:
If the socket has been explicitly passed in from outside I don't think we can guarantee that the socket is nonblocking, right?

Reply:
This is inside a fileprivate method that is only called internally with the socket from accept:

https://github.com/apple/swift-nio/blob/master/Sources/NIO/SocketChannel.swift#L1066